sql/core/src/test/scala/org/apache/spark/sql/collation/CollationTypePrecedenceSuite.scala

```diff
@@ -175,6 +175,115 @@ class CollationTypePrecedenceSuite extends QueryTest with SharedSparkSession {
     )
   }
 
+  test("in subquery expression") {
+    val tableName = "subquery_tbl"
+    withTable(tableName) {
+      sql(
+        s"""
+           |CREATE TABLE $tableName (
+           | c1 STRING COLLATE UTF8_LCASE,
+           | c2 STRING COLLATE UNICODE
+           |) USING $dataSource
+           |""".stripMargin)
```

Comment:
indentation should be 2 spaces not 3. Please, fix here and below:
```suggestion
          |CREATE TABLE $tableName (
          | c1 STRING COLLATE UTF8_LCASE,
          | c2 STRING COLLATE UNICODE
          |) USING $dataSource
          |""".stripMargin)
```